### PR TITLE
Initial wolfMQTT Espressif ESP32 template and AWS IoT examples

### PR DESCRIPTION
**Pull request by gojimmypi**
Adds initial Espressif ESP32 examples for either stand-alone or managed components as published at the [ESP Registry](https://components.espressif.com/components/wolfssl/wolfmqtt).

See also the development [ESP Registry Staging Site](https://components-staging.espressif.com/components/gojimmypi/mywolfmqtt).

Code is published with the scripts `/espressif/wolfssl_component_publish.sh [path]` script.

* edit: Add some CMake / VisualGDB / Espressif file exclusions to `.gitignore`

See https://github.com/wolfSSL/wolfssl/issues/6234 for a roadmap of Espressif improvements.

### CODE REVIEW

**Review by dgarske — APPROVED** [review]
Over to @embhorn

File: IDE/Espressif/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# IDE: Espressif
+
+Here are the libraries and features related to Espressif and the wolfMQTT product.
+
+For details on wolfMQTT [see the wolfMQTT Manual](https://www.wolfssl.com/documentation/manuals/wolfmqtt/wolfMQTT-Manual.pdf).
+
+## Component Manager
+
+[These files](./component-manager/README.md) are use internally by wolfSSL to publish components to the [Espressif ESP Registry](https://components.espressif.com/).
```
Comment (embhorn) [inline IDE/Espressif/README.md:9]: are **USED** internally

**Review by embhorn — APPROVED** [review]
Not a blocker for this PR, but would it be feasible to add a GitHub workflow to CI test the ESP32 build?

File: IDE/Espressif/README.md (hunk 1 above)
Comment (gojimmypi) [reply IDE/Espressif/README.md]: Good catch on the readme file! I've made that correction & also added some updates to `.gitignore`.

**Comment by gojimmypi** [comment]
> would it be feasible to add a GitHub workflow to CI test the ESP32 build?

Yes, I've been wanting to do just that. I'll reach out to @bandi13 and coordinate that.

**Comment by gojimmypi** [comment]
I've added https://github.com/wolfSSL/wolfMQTT/pull/390 that implements the most basic of ESP32 build tests.

**Pull request merged**